src/include/miopen/conv/invokers/cellfft.hpp

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+*
+* MIT License
+*
+* Copyright (c) 2020 Advanced Micro Devices, Inc.
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*
+*******************************************************************************/
+// clang-format off
+#ifndef GUARD_MIOPEN_CELLFFT_HPP
+#define GUARD_MIOPEN_CELLFFT_HPP
+
+#include <miopen/invoker.hpp>
+#include "../../../../solver/cellfft/include/cellfft_param.hpp"
```

Comment:
Can you move cellfft_param.hpp to the include directory so you won't have to use `../` stuff?

Reply:
I think i can move it into solver/ directory because this file only used by cellfft?

Reply:
You can make it `#include <cellfft/param.hpp>` if you wish to distinguish it from other headers even more. And it has cellfft in its name anyway :)

Reply:
Ok..., I'll wait for other people's review results to save CI resources

Reply:
The cellft_param module (.hpp/.cpp) is used only in cellfft.cpp. What is the purpose of detaching of cellfft_param?

Reply:
@atamazov :
The reason for the separation is that when the optimal solver is selected in the initialization of a model, the host parameters of each layer can be cached without repeated calculation. Although it is uncertain whether miopen will support it, this possibility is not ruled out

Reply:
If I understand you correctly, they effectively get cached inside of an invoker.

Reply:
> The reason for the separation is that when the optimal solver is selected in the initialization of a model, the host parameters of each layer can be cached without repeated calculation. Although it is uncertain whether miopen will support it, this possibility is not ruled out

Right now entities declared in celfft_param.hpp are used only in one module, and there is no need to detach it. Let's put it all together. This wouldn't cut off future possibilities. When/if it is needed somewhere else, we can easily refactor, -- by moving the necessary functionality into separate module.

Reply:
Usage of `../../..` stuff in #include directives is not allowed anyway.

Reply:
> > The reason for the separation is that when the optimal solver is selected in the initialization of a model, the host parameters of each layer can be cached without repeated calculation. Although it is uncertain whether miopen will support it, this possibility is not ruled out
> 
> Right now entities declared in celfft_param.hpp are used only in one module, and there is no need to detach it. Let's put it all together. This wouldn't cut off future possibilities. When/if it is needed somewhere else, we can easily refactor, -- by moving the necessary functionality into separate module.

Ok, i`ll changed and do same for flexgemm

Reply:
Good!